### PR TITLE
gadget: add new Kernel{Info,Asset} struct and helpers

### PR DESCRIPTION
**Pull request by mvo5**
This is the start of implementing the kernel assets spec as
outlined in: https://forum.snapcraft.io/t/415/46

### CODE REVIEW

File: gadget/kernel.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+)
+
+// XXX: should this be in a "gadget/kernel" or "kernel" package?
```
Comment (pedronis) [inline gadget/kernel.go:30]: a kernel package seems the right thing (we might move some bits from bootloader or boot there at some point if we can tame the likely ensuing circular import problems)

File: gadget/kernel.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	return &ki, nil
+}
+
+// ReadInfo reads the kernel specific metadata from meta/kernel.yaml
+// in the snap root directory.
+func ReadKernelInfo(kernelSnapRootDir string) (*KernelInfo, error) {
+	p := filepath.Join(kernelSnapRootDir, "meta", "kernel.yaml")
+	content, err := ioutil.ReadFile(p)
+	if err != nil {
```
Comment (bboozzoo) [inline gadget/kernel.go:57]: should we return a nil or empty KernelInfo when there's no meta/kernel.yaml?

Comment (pedronis) [reply gadget/kernel.go:57]: I don't remember what we do but we should be consistent with what gadget is doing

File: gadget/kernel.go (hunk 1 above)
Comment (mvo5) [reply gadget/kernel.go]: Thanks, I moved this out now and also moved editionNumber from gadget into a shared gadget/edition package.

**Pull request closed without merging**